07-routing-gossip.md

```diff
@@ -753,15 +683,89 @@ one.  If there's no `channel_update` then it is not sent at all, which is most
+timestamp is to be used as the time of the `channel_announcement`, but new nodes on
+the network will not have this, and further would require the first `channel_update`
+timestamp to be stored. Instead, we allow any update to be used, which
+is simple to implement.
 
 In the case where the `channel_announcement` is nonetheless missed,
 `query_short_channel_ids` can be used to retrieve it.
 
+## Initial Sync
+
+If a node requires an initial sync of gossip messages, it will be flagged
+in the `init` message, via a feature flag ([BOLT #9](09-features.md#assigned-localfeatures-flags)).
+
+Note that the `initial_routing_sync` feature is overridden (and should
+be considered equal to 0) by the `gossip_queries` feature if the
+latter is negotiated via `init`.
+
+Note that `gossip_queries` does not work with older nodes, so the
+value of `initial_routing_sync` is still important to control
+interactions with them.
+
+### Requirements
+
+A node:
+  - if the `gossip_queries` feature is negotiated:
+	- MUST NOT relay any gossip messages unless explicitly requested.
+  - otherwise:
+    - if it requires a full copy of the peer's routing state:
+      - SHOULD set the `initial_routing_sync` flag to 1.
+    - upon receiving an `init` message with the `initial_routing_sync` flag set to
+    1:
+      - SHOULD send gossip messages for all known channels and nodes, as if they were just
+      received.
+    - if the `initial_routing_sync` flag is set to 0, OR if the initial sync was
+    completed:
+      - SHOULD resume normal operation, as specified in the following
+      [Rebroadcasting](#rebroadcasting) section.
+
+## Rebroadcasting
+
+### Requirements
+
+A receiving node:
+  - upon receiving a new `channel_announcement` or a `channel_update` or
+  `node_announcement` with an updated `timestamp`:
+    - SHOULD update its local view of the network's topology accordingly.
+  - after applying the changes from the announcement:
+    - if there are no channels associated with the corresponding origin node:
+      - MAY purge the origin node from its set of known nodes.
+    - otherwise:
+      - SHOULD update the appropriate metadata AND store the signature
+      associated with the announcement.
+        - Note: this will later allow the node to rebuild the announcement
+        for its peers.
+
+A node:
+  - if the `gossip_queries` feature is negotiated:
+	- MUST not send gossip until it receives `gossip_timestamp_range`.
+  - SHOULD flush outgoing gossip messages once every 60 seconds, independently of
+  the arrival times of the messages.
+    - Note: this results in staggered announcements that are unique (not
+    duplicated).
+  - MAY re-announce its channels regularly.
+    - Note: this is discouraged, in order to keep the resource requirements low.
+  - upon connection establishment:
+    - SHOULD send all `channel_announcement` messages, followed by the latest
+    `node_announcement` AND `channel_update` messages.
+
```

Comment:
I think this section ("upon connection establishment") should just be removed: as is it is wrong, and what nodes should do when they receive the `init` message is already described in this document.

Reply:
hi @sstone this section has not been modified in this PR, it's merely been moved such that the section ordering matches the ordering listed in the Table of Contents for this BOLT spec. if we're going to delete it, we should merge this PR in first (to fix the ordering) and then edit the text to be correct.